### PR TITLE
AAP-7555 - Building the Getting Started with Event-Driven Ansible

### PR DESCRIPTION
**Pull request by jself-sudoku**
Set-up of all the symlinks and relevant directories: 

- attributes
- aap-common
- modules directory (eda)
- images

Added content to 4 chapters (assemblies and modules):

1. Event-Driven Ansible automation
2. Ansible rulebooks
3. Installation of Event-Driven controller
4. Using Event-Driven Ansible controller

### CODE REVIEW

File: downstream/modules/eda/con-decision-frameworks.adoc [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+= Decision Frameworks
+
+
+{EDAName} includes, by default, an ansible.eda collection, which contains some sample sources, event filters and rulebooks. All the collections, ansible rulebooks and their dependencies use a Decision Environment, which is an image that can be run on either Podman or Kubernetes.
+
+In Decision Environments, sources, which are typically Python code, are distributed through ansible-collections. They inject external events into a rulebook for processing. The rulebook consists of the following:
+
+* The python interpreter
+* Java Runtime Environment
```
Comment (benthomasson) [inline downstream/modules/eda/con-decision-frameworks.adoc:11]: We should say why we require Java here:   "Java Runtime Environment for Drools rule engine"

**Review by benthomasson — DISMISSED** [review]

File: downstream/assemblies/eda/assembly-ansible-rulebooks.adoc [hunk 2]
```diff
@@ -0,0 +1,18 @@
+
+[id="ansible-rulebooks"]
+
+= Ansible rulebooks
+
+{EDAcontroller} provides the interface in which Event-Driven Ansible automation performs. The Ansible rulebook provides the framework for Event-Driven Ansible automation. Ansible rulebook is essentially a collection of rulesets, which in turn, consists of one or more sources, rules, and conditions.
```
[the remark this answers is not in the document]
Comment (jself-sudoku) [reply downstream/assemblies/eda/assembly-ansible-rulebooks.adoc:6]: Thanks, Ben. I had it that way originally and changed it. Will revert back to the original.

File: downstream/modules/eda/con-decision-frameworks.adoc [hunk 3]
```diff
@@ -0,0 +1,15 @@
+[id="decision-frameworks"]
+
+= Decision Frameworks
```
[the remark this answers is not in the document]
Comment (jself-sudoku) [reply downstream/modules/eda/con-decision-frameworks.adoc:3]: Updated.

File: downstream/modules/eda/con_rulebook-actions.adoc [hunk 4]
```diff
@@ -0,0 +1,15 @@
+[id="rulebook-actions_{context}"]
+
+= Rulebook actions
+
+A rulebook specifies actions to be performed when a rule is triggered. A rule gets triggered when the events match the conditions for the rules. The following actions are currently supported:
+
+* `run_job_template`
+* `run_playbook`
```
Comment (Alex-Izquierdo) [inline downstream/modules/eda/con_rulebook-actions.adoc:8]: Should this action be removed if we are not officially supporting it in downstream? 
Please confirm @benthomasson @josephpisciotta

File: downstream/assemblies/eda/assembly-installation-eda-controller.adoc
Comment (hherbly) [inline downstream/assemblies/eda/assembly-installation-eda-controller.adoc:1]: Jameria, I'm confused about this sentence in line 7: 
The installation of Event-Driven Ansible controller coincides with other Ansible Automation Platform components.
Is "coincides" the right verb to use there? I'm not sure what you mean. Does the installation rely on or use other AAP components?

File: downstream/modules/eda/con-decision-environments.adoc
Comment (hherbly) [inline downstream/modules/eda/con-decision-environments.adoc:1]: I think I see an extra space in line 16 right before the word Dockerfile. Also, in the same line, is the word "optionally" necessary? I'm not sure it adds anything.

File: downstream/modules/eda/ref-deploy-eda-controller-with-aap-operator-on-ocp.adoc
Comment (hherbly) [inline downstream/modules/eda/ref-deploy-eda-controller-with-aap-operator-on-ocp.adoc:1]: In line 5, you left out a "the"--it should be "follow _the_ instructions..." etc.

**Comment by hherbly** [comment]
I left a couple of comments addressing clarity/consistency but once those are fixed, everything looks good to me. Let me know if you have any questions!

File: downstream/modules/eda/ref-deploy-eda-controller-with-aap-operator-on-ocp.adoc [hunk 5]
```diff
@@ -0,0 +1,7 @@
+[id="deploying-eda-controller-with-aap-operator-on-ocp"]
+
+= Deploying {EDA controller} with {OperatorPlatform} on {OCPShort}
+
+{EDAName} is not limited to {PlatformNameShort} on VMs. You can also access this feature on  {OperatorPlatform} on {OCPShort}. To deploy install {EDAName} with {OperatorPlatform}, follow instructions in link:https://access.redhat.com/documentation/en-us/red_hat_ansible_automation_platform/2.3/html-single/deploying_the_red_hat_ansible_automation_platform_operator_on_openshift_container_platform/index#deploy-eda-controller-on-aap-operator-ocp[Deploying Event-Driven Ansible controller with Ansible Automation Platform Operator on OpenShift Container Platform].
```
Comment (dcdacosta) [inline downstream/modules/eda/ref-deploy-eda-controller-with-aap-operator-on-ocp.adoc:5]: Remove extra space ..."feature on  {OperatorPlatform}"

File: downstream/titles/eda/eda-getting-started-guide/docinfo.xml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<title>Getting Started with Event-Driven Ansible</title>
```
Comment (dcdacosta) [inline downstream/titles/eda/eda-getting-started-guide/docinfo.xml:1]: This title must match the title in the master.adoc exactly for Pantheon to process it without errors.
Suggest you add Guide here or remove it from the master.adoc file; your choice. ;o)

File: downstream/modules/eda/con-decision-environments.adoc [hunk 7]
```diff
@@ -0,0 +1,15 @@
+[id="decision-environments"]
+
+= Decision Environments
+
+
+{EDAName} includes, by default, an ansible.eda collection, which contains some sample sources, event filters and rulebooks. All the collections, ansible rulebooks and their dependencies use a Decision Environment, which is an image that can be run on either Podman or Kubernetes.
```
Comment (dcdacosta) [inline downstream/modules/eda/con-decision-environments.adoc:6]: Suggest removing the word "some", I don't think it's necessary. It's cleaner to state:
EDA includes, by default, an ansible.eda collection, which contains sample sources, event filters and rulebooks.

**Review by dcdacosta — COMMENTED** [review]
A few minor comments, otherwise, you did a great job on this!

File: downstream/modules/eda/ref-deploy-eda-controller-with-aap-operator-on-ocp.adoc [hunk 8]
```diff
@@ -0,0 +1,7 @@
+[id="deploying-eda-controller-with-aap-operator-on-ocp"]
+
+= Deploying {EDA controller} with {OperatorPlatform} on {OCPShort}
```
Comment (dcdacosta) [inline downstream/modules/eda/ref-deploy-eda-controller-with-aap-operator-on-ocp.adoc:3]: Remove space in attribute so it renders correctly. {EDAcontroller}

File: downstream/assemblies/eda/assembly-installation-eda-controller.adoc [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+
+= Installation of {EDAname} controller
+
+The installation of Event-Driven Ansible controller coincides with other Ansible Automation Platform components. Like automation controller and automation hub, the setup includes default settings for specific variables in the inventory files.
+
+include::eda/ref-installing-eda-controller-on-aap.adoc[leveloffset=+1]
+include::eda/ref-deploy-eda-controller-with-aap-operator-on-ocp.adoc[leveloffset=+1]
+
+[role="_additional-resources"]
+== Additional resources (or Next steps)
```
Comment (dcdacosta) [inline downstream/assemblies/eda/assembly-installation-eda-controller.adoc:13]: Change this to a . heading to be consistent with rest of the doc:
.Additional resources (or Next steps)

File: downstream/assemblies/eda/assembly-installation-eda-controller.adoc
Comment (jself-sudoku) [reply downstream/assemblies/eda/assembly-installation-eda-controller.adoc:1]: Yes, Event-Driven Ansible requires the installation of automation controller. Customers have to ensure they have that installed first, per SMEs Joe Pisciotta and Ben Thomasson. Does this help to clarify things? 

The prerequisite for this is stated as an admonition (Important note) in the new section, "2.1.16 Single Event-Driven Ansible controller node with internal database". Does this clear this up?

Comment (hherbly) [reply downstream/assemblies/eda/assembly-installation-eda-controller.adoc:1]: Yes, that makes sense, and I think those changes would help. Thanks!

File: downstream/modules/eda/con-decision-environments.adoc
Comment (jself-sudoku) [reply downstream/modules/eda/con-decision-environments.adoc:1]: In this instance, the word "optionally" is used to let the customer know they have the option of deploying their CA certificate. I don't see anything wrong with using this word in this case. It's rather subjective, but I don't see that as a hindrance to the meaning of the sentence. So I'm leaving it as is for now.

File: downstream/modules/eda/con-decision-environments.adoc (hunk 7 above)
Comment (jself-sudoku) [reply downstream/modules/eda/con-decision-environments.adoc:6]: Agree in this instance. Removing the word.

File: downstream/modules/eda/con_rulebook-actions.adoc (hunk 4 above)
Comment (jself-sudoku) [reply downstream/modules/eda/con_rulebook-actions.adoc:8]: @Alex-Izquierdo - I see your question has not been answered. I've posted a note in the #event-driven-ansible Slack channel and tagged Ben and Joe there with hope that we get an answer to this question soon.

File: downstream/modules/eda/ref-deploy-eda-controller-with-aap-operator-on-ocp.adoc
Comment (jself-sudoku) [reply downstream/modules/eda/ref-deploy-eda-controller-with-aap-operator-on-ocp.adoc:1]: done

File: downstream/modules/eda/ref-deploy-eda-controller-with-aap-operator-on-ocp.adoc (hunk 8 above)
Comment (jself-sudoku) [reply downstream/modules/eda/ref-deploy-eda-controller-with-aap-operator-on-ocp.adoc:3]: done

File: downstream/modules/eda/ref-deploy-eda-controller-with-aap-operator-on-ocp.adoc (hunk 5 above)
Comment (jself-sudoku) [reply downstream/modules/eda/ref-deploy-eda-controller-with-aap-operator-on-ocp.adoc:5]: done

File: downstream/titles/eda/eda-getting-started-guide/docinfo.xml (hunk 6 above)
Comment (jself-sudoku) [reply downstream/titles/eda/eda-getting-started-guide/docinfo.xml:1]: done

File: downstream/titles/eda/eda-getting-started-guide/master.adoc [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+:experimental:
+
+include::attributes/attributes.adoc[]
+
+
+// Book Title
+= Getting Started with {EDAname} Guide
+
+Thank you for your interest in {EDAname}. {EDAname} is a new way to enhance and expand automation. It helps teams automate decision-making and improve IT speed and agility .
```
Comment (dcdacosta) [inline downstream/titles/eda/eda-getting-started-guide/master.adoc:14]: @jself-sudoku There is a blank space between agility and the period ending the sentence. I didn't see it until I formatted the doc locally to verify a clean build.

**Review by dcdacosta — APPROVED** [review]
Minor fix to remove a blank space, otherwise, LGTM!

File: downstream/modules/eda/con_rulebook-actions.adoc (hunk 4 above)
Comment (jself-sudoku) [reply downstream/modules/eda/con_rulebook-actions.adoc:8]: @Alex-Izquierdo - for now, I have commented out this line until we hear back from @benthomasson and @josephpisciotta .

**Review by hherbly — APPROVED** [review]
LGTM!

**Review by jself-sudoku — COMMENTED** [review]
Update to the second bullet changing it from "Java Runtime Environment" to "Java Runtime Enviroment for Drools rule engine" was applied, per request from Architect @benthomasson .

**Pull request merged**